src/sentry/static/sentry/app/views/stream/savedSearchSelector.jsx

```diff
@@ -222,12 +223,16 @@ const SavedSearchSelector = withApi(
     };
 
     getTitle() {
-      let searchId = this.props.searchId || null;
-      if (!searchId) return t('Custom Search');
-      let results = this.props.savedSearchList.filter(search => {
-        return searchId === search.id;
-      });
-      return results.length ? results[0].name : t('Custom Search');
+      let {searchId, query, savedSearchList} = this.props;
+      let result;
+
+      if (searchId) {
+        result = savedSearchList.find(search => searchId === search.id);
+      } else {
+        result = savedSearchList.find(search => query === search.query);
```

Comment:
Does `query` come from the search input? If so, I know it sometimes does funky things with whitespace, do we want to trim it?

Reply:
It comes from the URL querystring at the top-level and snakes its way down to here. From testing on my saved searches I didn't need to trim. I can check again when this is deployed.